executor/platform_compat/windows/pty_manager.py

```
@@ -0,0 +1,192 @@
+        self._pty = pty_process
+        self._rows = rows
+        self._cols = cols
+        self._returncode: Optional[int] = None
+
+    def read(self, size: int = 4096) -> bytes:
+        """Read data from the PTY."""
+        try:
+            data = self._pty.read(size)
+            if data:
+                return data.encode("utf-8") if isinstance(data, str) else data
+            return b""
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)